cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -883,19 +883,29 @@ private String queryByLike(String index, boolean hasIndex, IndexTextAndJsonValue
       sql = arrayNode(index, node, modifiers, relationModifiers, schemaIndex, vals, targetTable);
     } else {
       Index schemaIndex = null;
+      Index otherIndex = null;
       if (targetTable != null) {
         schemaIndex = DbSchemaUtils.getIndex(index, targetTable.getGinIndex());
+        otherIndex = DbSchemaUtils.getIndex(index, targetTable.getIndex());
       }
       String likeOperator = comparator.equals("<>") ? " NOT LIKE " : " LIKE ";
-      String like = "'" + Cql2SqlUtil.cql2like(node.getTerm()) + "'";
+      String term = "'" + Cql2SqlUtil.cql2like(node.getTerm()) + "'";
+      String indexMod;
       if(schemaIndex != null && schemaIndex.getMultiFieldNames() != null) {
-        sql = wrapInLowerUnaccent(schemaIndex.getFinalSqlExpression(targetTable.getTableName()), schemaIndex);
+        indexMod = wrapInLowerUnaccent(schemaIndex.getFinalSqlExpression(targetTable.getTableName()), schemaIndex);
       } else if(schemaIndex != null && schemaIndex.getSqlExpression() != null) {
-        sql = schemaIndex.getSqlExpression();
+        indexMod = schemaIndex.getSqlExpression();
       } else {
-        sql = wrapInLowerUnaccent(indexText, schemaIndex);
+        indexMod = wrapInLowerUnaccent(indexText, schemaIndex);
+      }
+      if(otherIndex != null) {
+        indexMod = wrapForLength(indexMod);
+        sql = "CASE WHEN length(" + term + ") <= 600 THEN "
+         + indexMod +  likeOperator + wrapForLength(wrapInLowerUnaccent(term, schemaIndex));
+        sql = appendLengthCase(likeOperator,likeOperator, indexText, term, comparator.equals("<>"), sql);
```

Comment:
The methods queryByLike and queryBySql contain almost the same code for the CASE...END clause. Can you extract the code into a method, for example
`private String caseLength(String operator, String index, String term)`
and let this method return the complete CASE WHEN ... THEN ... ELSE ... END clause?
This will also improve readability because the complete CASE ... END clause is at one place and is no longer split over several methods.

Reply:
I extracted these 2 functions but they are different enough that I cannot merge them yet.  I need to look at this closer.  I can do this if we want to wait on this PR.